grakn-graph/src/main/java/ai/grakn/graph/internal/InstanceImpl.java

```diff
@@ -88,26 +92,20 @@
      */
     @Override
     public void delete() {
-        InstanceImpl<?, ?> parent = this;
-        Set<CastingImpl> castings = parent.castings();
+        Set<Relation> relations = castingsInstance().map(Casting::getRelation).collect(Collectors.toSet());
```

Comment:
This is picky, but there's no need to collect these. You can just stream over them.